Tests/UnitTests/Core/EventData/MultiTrajectoryTests.cpp

```diff
@@ -868,4 +868,19 @@ BOOST_AUTO_TEST_CASE(ProxyAssignment) {
   // MultiTrajectory::TrackStateProxy tp4{tp3};
 }
 
+// Check if the copy from const does compile, assume the copy is done correctly
+BOOST_AUTO_TEST_CASE(CopyFromConst) {
+  using PM = TrackStatePropMask;
+  MultiTrajectory mj;
+
+  const auto idx_a = mj.addTrackState(PM::All);
+  const auto idx_b = mj.addTrackState(PM::All);
+
+  auto mutableProxy = mj.getTrackState(idx_a);
+  auto constProxy =
+      static_cast<const MultiTrajectory&>(mj).getTrackState(idx_b);
```

Comment:
Does a static cast do the right thing here? I'd probably created an auxiliary variable `const MultiTrajectory&` to test this.

Reply:
It should be equivalent, but maybe its better to be more explicit here